snuba/datasets/discover.py

```diff
@@ -53,16 +53,17 @@ def detect_table(
     """
     event_columns = set()
     transaction_columns = set()
+    using_duration = False
     for col in query.get_all_referenced_columns():
         if events_only_columns.get(col):
             event_columns.add(col)
         elif transactions_only_columns.get(col):
             transaction_columns.add(col)
+            using_duration = using_duration or col == "duration"
```

Comment:
wait though, isn't this going to fail the queries the other way around ?
if the query was referencing the duration column and no event-exclusive columns, this should have already resolved to the transaction table, right?
But if we were having inconsistencies it was because we had both transactions columns and events columns, so isn't this just hiding the problem by making us return invalid information from the events table?

Reply:
Yes, this is a stopgap measure to cut down on the exceptions caused. It won't fully address the core issue. Ultimately, we will always return invalid information from one table or another.

Reply:
well, ok let's see which one is the conflicting column if we switch to transactions for this.I have the suspect iut is going to be the group id, which is added by the same bug we are trying to solve separately.

Reply:
Yeah that might make more sense. OK, I'll put this on hold.

Reply:
#930 has been active for a few days, how would you like to proceed ?

Reply:
I'll close this. Currently it appears the only way to trigger this is to explicitly search for error events, with a duration column/function. The errors are WAY down so I think this isn't necessary.